source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c

```diff
@@ -335,16 +348,12 @@ static void prvEMACDeferredInterruptHandlerTask( void * pvParameters )
                     #if ( ipconfigDRIVER_INCLUDED_RX_IP_CHECKSUM == 1 )
                     {
                         /* the Atmel SAM GMAC peripheral does not support hardware CRC offloading for ICMP packets.
-                         * It must therefore be implemented in software. */
-                        pxIPPacket = ( IPPacket_t const * ) pxBufferDescriptor->pucEthernetBuffer;
-
-                        if( pxIPPacket->xIPHeader.ucProtocol == ( uint8_t ) ipPROTOCOL_ICMP )
-                        {
+                        * It must therefore be implemented in software. */
+                        if ( isICMP(pxBufferDescriptor) ) {
```

Comment:
The `if` statement expects a boolean value. The function `isICMP()` returns an integer value. The only way to satisfy the compiler is as follows:
~~~
if( isICMP( pxBufferDescriptor ) == pdTRUE )
{
}
~~~